modules/standard/Path.chpl

```diff
@@ -286,4 +286,80 @@ proc file.realPath(): string throws {
       return false;
     }
   }
+
+  /* Determines and returns the longest common path prefix of
+   all the string pathnames provided. If no arguments are specified, returns
+   the empty string.
+
+   :arg paths: Any number of paths as an array
+   :type paths: `array`
+
+   :return: The longest common path prefix
+   :rtype: `string`
+*/
+
+  proc commonPrefix(paths: []): string {
+
+  var result: string = "";    // result string
+  var inputLength = paths.size;   // size of input array
+  var start: int = paths.domain.first;
+  var end: int = paths.domain.last;
+  var firstPath = paths[start];
+  var delimiter: string;
+  var minimum: int;
+  var flag: int = 0;
+
+  // if input is empty, return empty string.
+  // if input is just one string, return that string as longest common prefix path.
+
+  if inputLength == 0 then {
+    return result;
+  } else if inputLength == 1 then{
+    return firstPath;
+  }
+
+  // finding delimiter to split the paths.
+
+  if firstPath.find("\\", 1..firstPath.length) == 0 then {
+    delimiter = "/";
+  } else {
+    delimiter = "\\";
+  }
+
+  var prefixArray = firstPath.split(delimiter, -1, false);    // array of resultant prefix string
+
+  var pos = prefixArray.size;   // rightmost index of common prefix
+  var minPathLength = prefixArray.size;
+
+  for i in (start+1)..end do {
+
+    var tempArray = paths[i].split(delimiter, -1, false);   // temporary array storing the current path under consideration
+
+    minimum = min(prefixArray.size, tempArray.size);
+
+    if minimum < minPathLength then {
+      minPathLength = minimum;
+    }
+
+    for (itr, ind) in zip(1..minimum, 1..minimum) do {
```

Comment:
Sorry I didn't catch this earlier, but `itr` and `ind` should always be the same value, so you don't need both of them to access into tempArray and prefixArray, just one.  I think it was a hold-over from when you were iterating over the contents of those arrays, which would be different.